lib/flight_job/commands/create_script.rb

```diff
@@ -32,42 +32,236 @@
-      def run
-        # Resolves the answers
-        answers = answers_input || begin
-          if $stdout.tty? && stdin_notes?
-            raise InputError, <<~ERROR.chomp
-              Cannot prompt for the answers as standard input is in use!
-              Please provide the answers with the following flag: #{pastel.yellow '--answers'}
-            ERROR
-          elsif $stdout.tty?
-            prompt_answers
-          else
-            msg = <<~WARN.chomp
-              No answers have been provided! Proceeding with the defaults.
+      MULTI_HELP = "(Press ↑/↓/←/→ arrow to scroll, Space/Ctrl+A|R to select (all|rev) and Enter to finish)"
+      SUMMARY = ERB.new(<<~'TEMPLATE', nil, '-')
+        <%= pastel.bold.underline 'SUMMARY' %>
+        <% questions.each do |question| -%>
+        <% next unless asked[question.id] -%>
+        <%= pastel.bold(question.text) %>
+        <%
+          value = answers[question.id]
+          value = (value.is_a?(Array) ? value.join(',') : value.to_s)
+        -%>
+        <%= pastel.green value %>
+        <% end -%>
+        <%= pastel.bold('Notes') %>
+        <%= pastel.green notes %>
+      TEMPLATE
+
+      # NOTE: The questions must be topologically sorted on their dependencies otherwise
+      # these prompts will not function correctly
+      QuestionPrompter = Struct.new(:pastel, :questions, :notes) do
+        # Initially set to the defaults
+        def answers
+          @answers ||= questions.map { |q| [q.id, q.default] }.to_h
+        end
+
+        def summary
+          SUMMARY.result self.binding
+        end
+
+        # Checks the questions dependencies and return if it should be prompted for
+        def prompt?(question)
+          return true unless question.related_question_id
+          answers[question.related_question_id] == question.ask_when['eq']
+        end
+
+        # Ask all the questions in order
+        def prompt_all
+          questions.each do |question|
+            prompt?(question) ? prompt_question(question) : skip_question(question)
+          end
+        end
+
+        # Prompts the user for any answers they wish to change
+        # return [Boolean] if the user requested questions to be re-asked
+        def prompt_again
+          opts = { default: 'None', show_help: :always }
+          case prompt.select("Would you like to change your answers?", ['All', 'Selected', 'None'], **opts)
+          when 'All'
+            prompt_all
+            prompt_notes
+            true
+          when 'Selected'
+            puts(pastel.yellow(<<~WARN).chomp) if dependencies?
+              WARN: Some of the questions have dependencies on previous answers.
+              The exact question prompts may differ if the dependencies change.
             WARN
-            $stderr.puts pastel.red(msg)
-            FlightJob.logger.warn msg
-            {}
+            opts = { show_help: :always, echo: false, cycle: true, help: MULTI_HELP }
```

Comment:
`cycle` should be set to false.  I tried to get to the last question by holding down the down key and I didn't get there :-(

Reply:
https://github.com/openflighthpc/flight-job/pull/9/commits/eb6e5e64f72d760a987b5e270a912ff33238643f